src/components/PickerDay.vue

```diff
@@ -2,44 +2,51 @@
   <div class="picker-view">
     <slot name="beforeCalendarHeaderDay" />
     <PickerHeader
-      :config="headerConfig"
-      :next="nextMonth"
-      :previous="previousMonth"
+      v-if="showHeader"
+      :is-next-disabled="isNextDisabled"
+      :is-previous-disabled="isPreviousDisabled"
+      :is-rtl="isRtl"
+      :is-up-disabled="isUpDisabled"
+      @next="nextPage"
+      @previous="previousPage"
     >
       <span
-        :class="allowedToShowView('month') ? 'up' : ''"
+        :class="{ up: !isUpDisabled }"
         class="day__month_btn"
-        @click="showPickerCalendar('month')"
+        @click="$emit('set-view', 'month')"
       >
         {{ pageTitleDay }}
       </span>
       <slot slot="nextIntervalBtn" name="nextIntervalBtn" />
       <slot slot="prevIntervalBtn" name="prevIntervalBtn" />
     </PickerHeader>
-    <div :class="isRtl ? 'flex-rtl' : ''">
-      <span v-for="d in daysOfWeek" :key="d.timestamp" class="cell day-header">
-        {{ d }}
-      </span>
-      <span
-        v-for="day in days"
-        :key="day.timestamp"
-        class="cell day"
-        :class="dayClasses(day)"
-        @click="selectDate(day)"
-      >
-        {{ dayCellContent(day) }}
+    <div :class="{ 'flex-rtl': isRtl }">
+      <span v-for="(day, i) in daysOfWeek" :key="i" class="day-header">
```

Comment:
Using the index of an array as key is bad practice it can result in unexpected behaviour, we should stick with the timestamp

Reply:
Ok thanks, I wasn't aware of that. There's actually no timestamp in the `dayOfWeek` array, so I'll just change it to the `day`.